ncm-cron/src/main/pan/components/cron/schema.pan

```diff
@@ -38,12 +38,63 @@ type structure_cron_log = {
     'mode' ? string
 } with structure_cron_log_valid(SELF);
 
+@documentation{
+    Validate contents of cron fields
+}
+function valid_cron_field = {
+    if (ARGC == 4) {
+        test_value = ARGV[0];
+        lower_bound = ARGV[1];
+        upper_bound = ARGV[2];
+        text_regex = ARGV[3];
+
+        # Check that the field contains only valid characters
+        result = match(test_value, '^(?:[a-z0-9/*-,]+)$');
+        if (!result) error(format('"%s" contains invalid characters', test_value));
+
+        # Find all strings of digits and validate them against provided bounds
+        foreach(k; v; matches(test_value, '([0-9]+)')) {
+            i = to_long(v);
+            if (i < lower_bound) error(format('Value %d is below lower bound of %d', i, lower_bound));
+            if (i > upper_bound) error(format('Value %d is above upper bound of %d', i, upper_bound));
+            result = result && i >= lower_bound && i <= upper_bound;
```

Comment:
is the `i >= lower_bound && i <= upper_bound` really needed here? you can't get here otherwise (in which case `result` is not needed either)